python/ray/air/execution/resources/fixed.py

```diff
@@ -0,0 +1,160 @@
+from typing import Dict, List, Optional, Union
+
+from dataclasses import dataclass
+
+import ray
+from ray import SCRIPT_MODE, LOCAL_MODE
+from ray.air.execution.resources.request import (
+    ResourceRequest,
+    AcquiredResource,
+    RemoteRayObject,
+)
+from ray.air.execution.resources.resource_manager import ResourceManager
+from ray.util.annotations import DeveloperAPI
+
+
+# Avoid numerical errors by multiplying and subtracting with this number.
+# Compare: 0.99 - 0.33 = 0.65999... vs (0.99 * 1000 - 0.33 * 1000) / 1000 = 0.66
+_DIGITS = 100000
+
+
+@DeveloperAPI
+@dataclass
+class FixedAcquiredResource(AcquiredResource):
+    bundles: List[Dict[str, float]]
+
+    def annotate_remote_objects(
+        self, objects: List[RemoteRayObject]
+    ) -> List[Union[RemoteRayObject]]:
+        # If we have an empty head, we schedule the first object (the "head") with
+        # empty resources.
+        if self.resource_request.head_bundle_is_empty:
+            bundles = [{}] + self.resource_request.bundles
+        else:
+            bundles = self.resource_request.bundles
```

Comment:
If there is only 1 ResourceRequest type, why do we have separate AcquiredResource types?
``annotate_remote_objects()`` for these 2 AcquiredResource types also look exactly the same?

Reply:
They are not the same - one sets placement group options to the remote objects and one only regular options. The PG bundle also needs to schedule the potential empty head bundle into the first PG bundle.
They look similar but they are different, and if we had more backends, we'd likely have more acquired resource types.

The user never instantiates these themselves, they just get a `AcquiredResource` class back with an `annotate_remote_objects` method, so the interface is always the same.